velox/common/memory/MemoryAllocator.h

```diff
@@ -582,7 +584,7 @@ class MemoryAllocator : std::enable_shared_from_this<MemoryAllocator> {
   inline static std::atomic<uint64_t> totalLargeAllocateBytes_;
 
  private:
-  inline static std::mutex initMutex_;
+  inline static folly::RWSpinLock spinLock_;
```

Comment:
Let's use folly::SharedMutex instead of spinlock here? Only use spinlock for a very quick operation like update a set of counters. Not sure if we do some blocking operation on allocator creation in the future.